azkaban-common/src/main/java/azkaban/project/AzkabanProjectLoader.java

```diff
@@ -79,10 +79,53 @@
       final File archive, final String fileType, final User uploader, final Props additionalProps)
       throws ProjectManagerException {
     log.info("Uploading files to " + project.getName());
+    final Map<String, ValidationReport> reports;
+
+    // Since props is an instance variable of ProjectManager, and each
```

Comment:
I am not sure I follow the comment. Let's chat off line.

Reply:
Discussed offline. I understand it now.
In general, I think the Props are overused in the current code.